resources/sdk/purecloudjava/templates/ApiClient.mustache

```diff
@@ -549,7 +607,8 @@ public class ApiClient implements AutoCloseable {
         String reasonPhrase = response.getStatusReasonPhrase();
         Map<String, String> headers = response.getHeaders();
 
-        if (statusCode >= 200 && statusCode < 300) {
+        // If the response indicates that the access_token has expired, and we wish to refresh it transparently, we'll return from this method and let the caller handle it
+        if (statusCode >= 200 && statusCode < 300 || statusCode == 401 && shouldRefreshAccessToken) {
```

Comment:
How does this work? It appears that it's going to process the response as if it were successful when the response is 401, which should fail because it's going to parse the response expecting it to the success response type, but the actual response is an error body.

Reply:
That probably wasn't the cleanest solution. I'll remove the changes to this method and update the callers (getAPIResponse and getAPIResponseAsyncWithRetry) to handle the exception. They can request a new access token if the status code is 401 and shouldRefreshAccessToken is set, or rethrow the exception if those conditions aren't met